src/tree/index.tsx

```diff
@@ -0,0 +1,415 @@
+import { create, tsx, diffProperty } from '@dojo/framework/core/vdom';
+import { RenderResult } from '@dojo/framework/core/interfaces';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+import { createResourceMiddleware } from '@dojo/framework/core/middleware/resources';
+import theme from '@dojo/framework/core/middleware/theme';
+import focus from '@dojo/framework/core/middleware/focus';
+import { flat } from '@dojo/framework/shim/array';
+
+import { Keys } from '../common/util';
+import Icon from '../icon';
+import Checkbox from '../checkbox';
+import { ListItem } from '../list';
+
+import * as css from '../theme/default/tree.m.css';
+import LoadingIndicator from '../loading-indicator';
+
+export interface TreeNodeOption {
+	id: string;
+	parent: string;
+	value: string;
+	hasChildren: boolean;
+}
+
+export interface TreeProperties {
+	checkable?: boolean;
+	selectable?: boolean;
+	checked?: string[];
+	expanded?: string[];
+	initialChecked?: string[];
+	initialExpanded?: string[];
+	selectedNode?: string;
```

Comment:
I think that this should be renamed to `value` to align with the `onValue` callback